### DISCUSSION TITLE
Procedure for contributing many little doc changes

### DISCUSSION THREAD

**Discussion by TonyGravagno** [rspamd/rspamd#3731] [Q&A] [answered]
As with many projects, the **documentation site** for Rspamd has many small language issues, typos, and bits of missing or unclear detail. I'd like to provide minor corrections and enhancements, as I go through sections of the doc. Any section might receive multiple updates as I related information from one part of the site to others. What is the preferred approach for accepting and processing doc changes like this?

At this time I'm not qualified for any major doc contribution but after going through it many times, eventually I might have something helpful to contribute as new content. For now I'd like to start small.

I believe the best way to do this for random changes is to provide multiple small PRs with a small number of focused commits in each. For something consistent across the site, I believe a single PR is appropriate with a number of related commits.

I propose that I create a single Issue for all of my changes, where I document issues in comments and then submit multiple PRs against that one ticket. The ticket can be closed after some number of months or PRs and the process can be repeated.

Thanks.

EDIT: Here are the kind of notes that I'm creating now, for actual changes as time permits:
https://github.com/TonyGravagno/rspamd.com/projects/1

**Comment by moisseev** [comment, marked as answer]
Issues and PRs are not tracked in the git log. Combining multiple commits in a PR is convenient for reviewing and discussing changes, but when the PR is approved, it is merged as a bunch of separate commits.

> I believe the best way to do this for random changes is to provide multiple small PRs with a small number of focused commits in each.

Yes, if the changes are unrelated, it is easier to read and understand them in separate commits, but if you do multiple similar changes it is better to make them in the same commit.

> For something consistent across the site, I believe a single PR is appropriate with a number of related commits.

When it is possible commits should be atomic. If some changes depend on another changes, you should make these related changes in one commit. This ensures that the site is always left in a consistent state, they are easier to track in the git log or revert.


> I propose that I create a single Issue for all of my changes, where I document issues in comments and then submit multiple PRs against that one ticket. The ticket can be closed after some number of months or PRs and the process can be repeated.

It is not necessary. You can submit PRs without opening a ticket and write those comments in the PRs.
